lib/model/query/submissions.js

```diff
@@ -386,14 +398,87 @@ const getForExport = (formId, instanceId, draft, options = QueryOptions.none) =>
   maybeOne(_export(formId, draft, [], options.withCondition({ 'submissions.instanceId': instanceId })))
     .then(map(_exportUnjoiner));
 
+////////////////////////////////////////////////////////////////////////////////
+// DELETE SUBMISSION
+
+const del = (submission) => ({ run }) =>
+  run(markDeleted(submission));
+
+del.audit = (submission, form) => (log) => log('submission.delete', submission.with({ acteeId: form.acteeId }), { instanceId: submission.instanceId });
+
+const restore = (submission) => ({ run }) =>
+  run(markUndeleted(submission));
+restore.audit = (submission, form) => (log) => log('submission.restore', { acteeId: form.acteeId }, { instanceId: submission.instanceId });
+
+////////////////////////////////////////////////////////////////////////////////
+// PURGING SOFT-DELETED SUBMISSIONS
+
+const DAY_RANGE = config.has('default.taskSchedule.purge')
+  ? config.get('default.taskSchedule.purge')
+  : 30; // Default is 30 days
+
+
+// Submission purging and the trash filter can target a specific submission
+// or all deleted submissions, but can't be used to filter-purge by project or form.
+const _trashedFilter = (force, projectId, xmlFormId, instanceId) => {
+  const idFilter = ((instanceId != null) && (projectId != null) && (xmlFormId != null)
+    ? sql`and submissions."instanceId" = ${instanceId}
+    and forms."projectId" = ${projectId}
+    and forms."xmlFormId" = ${xmlFormId}`
+    : sql``);
+  return (force
+    ? sql`submissions."deletedAt" is not null ${idFilter}`
+    : sql`submissions."deletedAt" < current_date - cast(${DAY_RANGE} as int) ${idFilter}`);
+};
+
+const purge = (force = false, projectId = null, xmlFormId = null, instanceId = null) => ({ oneFirst }) => {
+  if ((instanceId != null || projectId != null || xmlFormId != null) && !(instanceId != null && projectId != null && xmlFormId != null)) {
+    throw Problem.internal.unknown({ error: 'Must specify either all or none of projectId, xmlFormId, and instanceId' });
+  }
+  return oneFirst(sql`
+with redacted_audits as (
+    update audits set notes = ''
+    from submissions
+    join forms on submissions."formId" = forms.id
+    where (audits.details->>'submissionId')::int = submissions.id
+    and ${_trashedFilter(force, projectId, xmlFormId, instanceId)}
+  ), deleted_client_audits as (
+    delete from client_audits
+    using submission_attachments, submission_defs, submissions, forms
+    where client_audits."blobId" = submission_attachments."blobId"
+    and submission_attachments."submissionDefId" = submission_defs.id
+    and submission_attachments."isClientAudit" = true
+    and submission_defs."submissionId" = submissions.id
+    and submissions."formId" = forms.id
+    and ${_trashedFilter(force, projectId, xmlFormId, instanceId)}
+  ), purge_audits as (
+    insert into audits ("action", "loggedAt", "processed", "details")
+    select 'submission.purge', clock_timestamp(), clock_timestamp(), jsonb_build_object('submissions_deleted', "count")
```

Comment:
Check if this should be filtered out of non-verbose audit logs, too. No actee id. Check what forms.purge does?

Will need small frontend change, too.